source/postprocess/particle_density_histogram.cc

```diff
@@ -0,0 +1,268 @@
+/*
+  Copyright (C) 2016 - 2024 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/postprocess/particle_density_histogram.h>
+#include <aspect/particle/manager.h>
+
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/fe/fe_values.h>
+#include <deal.II/grid/tria_accessor.h>
+#include <cmath>
+
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+    template <int dim>
+    std::pair<std::string,std::string>
+    ParticleDensityStatistics<dim>::execute (TableHandler &statistics)
+    {
+      double  local_min_score = std::numeric_limits<double>::max();
+      double local_max_score = 0;
+      double global_score = 0;
+      unsigned int cells_with_particles = 0;
+
+
+      for (const typename Triangulation<dim>::active_cell_iterator &cell : this->get_dof_handler().active_cell_iterators())
+      {
+        if (cell->is_locally_owned())
+          {
+            /*
+              Create a table with perfect distribution (even number of particles in each bucket)
+              take the distance squared between the "perfect" table and the worst case table
+              Create a table with the actual distribution
+              Take distance between "perfect" table and the actual table
+              return the ratio between the observed distance and the worst case distance.
+              a value of 1 represents the worst possible distribution, a value of 0 represents a completely even distribution.
+            */
```

Comment:
Please be consistent with the capitalization of the sentences. I prefer capitalization of the first word, but the important point is that it is consistent.